### PR TITLE
feat(close-button): phase 3–4 API and accessibility

### PR DESCRIPTION
**Pull request by TarunAdobe**
Phase 3 API migration for swc-close-button. Builds on the Phase 2 scaffold PR into ttomar/migrate-close-button (SharedButtonBase, minimal render/CSS, no pending on close-button).

  • Finalize public API on CloseButtonBase extending SharedButtonBase: size, static-color, accessible-label, disabled
  • Breaking change: no variant on swc-close-button — use static-color only (1st-gen sp-close-button unchanged in this PR)
  • Add STATIC_COLORS on base and invalid static-color DEBUG validation (Divider pattern)
  • Expand JSDoc on base class and fenced @example blocks on CloseButton.ts
  • Add API-focused Storybook stories: Sizes, StaticColors, States
  • Add Storybook contract tests: defaults/reflection, slot accessible name, invalid static-color / missing accessible-label warnings, options-story smoke tests

  **Docs**

  • Tick API items in close-button migration plan (CONTRIBUTOR-DOCS/03_project-planning/03_components/close-button/migration-plan.md)

  **Out of scope (follow-up PRs on integration branch)**

  • S2 closebutton CSS / token fidelity
  • Playwright a11y snapshots, keyboard suite, VRT
  • Consumer migration-guide.mdx, changeset, final merge to main

  **Motivation and context**

  Close-button migration follows the washing-machine Phase 3 — API step: lock the 2nd-gen public surface, add DEBUG validation, and prove the contract with tests
  before styling and full a11y work land.

  This PR targets the integration branch (ttomar/migrate-close-button), not main. Migration phases merge there first; one final merge to main when the component is complete.

  **Related issue(s)**

  • SWC-2087 (epic — close-button 2nd-gen migration)
  • SWC-2089 (planning)

  Depends on the scaffold PR in ttomar/migrate-close-button (stacked review is fine).

  **Screenshots (if appropriate)**

  N/A — no visual fidelity changes beyond existing scaffold stub CSS.

  ────────────────────────────────────────

  **Author's checklist**

  ◼ I have read the CONTRIBUTING (https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md) and PULL_REQUESTS 
    (https://github.com/adobe/spectrum-web-components/blob/main/PULL_REQUESTS.md) documents.
  ◼ I have reviewed the Accessibility Practices for this feature, see: Aria Practices (https://www.w3.org/TR/wai-aria-practices/)
  ◼ I have added automated tests to cover my changes.
  ◻ I have included a well-written changeset if my change needs to be published. (Deferred — integration branch; changeset with final merge to main)
  ◻ I have included updated documentation if my change required it. (Migration plan API checklist only; consumer guide in a later phase)

  ────────────────────────────────────────

  **Reviewer's checklist**

  ◻ Includes a Github Issue with appropriate flag or Jira ticket number without a link
  ◻ Includes thoughtfully written changeset if changes suggested include patch, minor, or major features (N/A for integration-branch PR)
  ◻ Automated tests cover all use cases and follow best practices for writing
  ◻ Validated on all supported browsers
  ◻ All VRTs are approved before the author can update Golden Hash

  **Manual review test cases**

  ◻ Public API (CEM / Storybook controls)
    1. Open 2nd-gen Storybook → Close Button → Playground.
    2. Confirm controls: size, static-color, accessible-label, disabled.
    3. Confirm no variant, pending, or pending-label on swc-close-button.

  ◻ Invalid static-color (DEBUG)
    1. Run with window.__swc.DEBUG = true.
    2. Open Close Button → Tests → Invalid Static Color Warning Test.
    3. Expect console warning listing allowed static-color values.

  ◻ API stories
    1. Sizes — four sizes render with distinct cross icons.
    2. StaticColors — default, white, and black on dark background.
    3. States — default and disabled; disabled control not activatable.

  ◻ Accessible name
    1. Tests → Overview Test — accessible-label → inner <button> aria-label.
    2. Slot Label Accessible Name Test — slot text Dismiss → aria-label="Dismiss".

  ◻ Build
    1. yarn workspace @spectrum-web-components/core build
    2. yarn workspace @adobe/spectrum-wc build
    3. yarn workspace @adobe/spectrum-wc test -- --run --project storybook components/close-button

  Device review

  ◻ Did it pass in Desktop?
  ◻ Did it pass in (emulated) Mobile?
  ◻ Did it pass in (emulated) iPad?

  **Accessibility testing checklist**

  Note: Full a11y phase is a follow-up PR. This PR adds API-level accessible-name tests; keyboard/focus uses the inner <button> from the scaffold.

  ◻ Keyboard (required — document steps below)
    1. Open 2nd-gen Storybook → Close Button → Overview.
    2. Tab to swc-close-button; focus should land on the inner button (delegated focus).
    3. Press Enter and Space — button activates (dismiss handler is author-owned).
    4. Open States; tab to disabled instance — skipped / not activatable via native disabled.
    5. Expect a visible focus indicator (stub CSS from scaffold).

  ◻ Screen reader (required — document steps below)
    1. Overview with accessible-label="Close" — announced as button, name "Close".
    2. Slot Label Accessible Name Test — name "Dismiss" from slot text.
    3. Icon must not duplicate the name (aria-hidden="true" on icon span).

### CODE REVIEW

**Comment by TarunAdobe** [comment]
Stacked on scaffold PR - this adds 2nd-gen API + Storybook contract tests only. Review CloseButton.base.ts, close-button.test.ts, and stories; confirm no variant / pending on swc-close-button. Styling and Playwright a11y are follow-ups.

**Review by blunteshwar — APPROVED** [review]
nit 
`CloseButton.VALID_SIZES` and `CLOSE_BUTTON_VALID_SIZES` resolve to the same list here. Could we use one access pattern throughout this story file for consistency?

File: 2nd-gen/packages/core/components/close-button/CloseButton.base.ts [hunk 1]
```diff
@@ -28,13 +33,22 @@ import {
  *
  * @attribute {CloseButtonSize} size - Visual size of the close button.
  * @attribute {'white' | 'black'} static-color - Static color treatment for display over colored or image backgrounds.
+ * @attribute {string} accessible-label - Accessible name when no visible label slot text is provided.
  * @attribute {boolean} disabled - Whether the button is disabled.
  */
 export abstract class CloseButtonBase extends ButtonBase {
   /** @internal */
   static override readonly VALID_SIZES: readonly CloseButtonSize[] =
```
Comment (miwha-adobe) [inline 2nd-gen/packages/core/components/close-button/CloseButton.base.ts]: Question: Is close button's size set intentionally independent of button's, or is it expected to always track button?

File: 2nd-gen/packages/swc/components/close-button/stories/close-button.stories.ts [hunk 2]
```diff
@@ -86,3 +91,122 @@ export const Overview: Story = {
...
 };
+
+// ──────────────────────────
+//    OPTIONS STORIES
+// ──────────────────────────
+
+export const Sizes: Story = {
+  render: (args) => html`
+    <div style="display: flex; align-items: center; gap: 16px;">
+      ${CLOSE_BUTTON_VALID_SIZES.map(
```
Comment (miwha-adobe) [inline 2nd-gen/packages/swc/components/close-button/stories/close-button.stories.ts]: I'm curious why here we use  `CLOSE_BUTTON_VALID_SIZES.map`, and on line 31, we use `CloseButton.VALID_SIZES`. I would think we should be consistent, and get our values off the class rather than the type constant.

File: 2nd-gen/packages/core/components/close-button/CloseButton.base.ts (hunk 1 above)
Comment (miwha-adobe) [reply 2nd-gen/packages/core/components/close-button/CloseButton.base.ts]: Meter had a good example of avoiding a redundant constant: https://github.com/adobe/spectrum-web-components/pull/6370/changes#diff-47fd3642df0bbbc6fb4195ef90cedd91dc2b6acb18c865446f73f9c86d0780d7

File: 2nd-gen/packages/swc/components/close-button/stories/close-button.stories.ts (hunk 2 above)
Comment (TarunAdobe) [reply 2nd-gen/packages/swc/components/close-button/stories/close-button.stories.ts]: yes i agree this was a miss. i fixed it thanks

File: 2nd-gen/packages/core/components/close-button/CloseButton.base.ts (hunk 1 above)
Comment (TarunAdobe) [reply 2nd-gen/packages/core/components/close-button/CloseButton.base.ts]: thanks for catching this. i have aligned it and removed CLOSE_BUTTON_VALID_SIZES and the VALID_SIZES override on CloseButtonBase.

File: 2nd-gen/packages/swc/components/close-button/stories/close-button.stories.ts [hunk 3]
```diff
@@ -86,3 +93,126 @@ export const Overview: Story = {
...
+
+      <section>
+        <h4 style="margin: 0 0 8px;">Visible slot label</h4>
+        <p style="margin: 0 0 12px;">
+          Slotted text is visually hidden but exposed as the button name when
+          <code>accessible-label</code>
+          is omitted.
+        </p>
+        <swc-close-button size=${args.size}>Dismiss</swc-close-button>
+      </section>
```
Comment (Rajdeepc) [inline 2nd-gen/packages/swc/components/close-button/stories/close-button.stories.ts]: Why did you bypass `template()` helper here? Future arg spreading capabilities are lost.

File: 2nd-gen/packages/swc/components/close-button/test/close-button.test.ts [hunk 4]
```diff
@@ -31,8 +42,437 @@ export default {
...
+    <swc-close-button accessible-label="Close"></swc-close-button>
+  `,
+  play: async ({ canvasElement, step }) => {
+    const closeButton = await getComponent<CloseButton>(
+      canvasElement,
+      'swc-close-button'
+    );
+
+    await step('warns when an invalid static-color is set in DEBUG mode', () =>
+      withWarningSpy(async (warnCalls) => {
```
Comment (Rajdeepc) [inline 2nd-gen/packages/swc/components/close-button/test/close-button.test.ts:426]: I am expecting the validation logic is in base branch, its not here right?

**Review by Rajdeepc — COMMENTED** [review]
Mostly looks good. can you also add a test verifying that accessible-label wins over slot text when both are provided. This is an edge case which is good to have as a test to avoid any regress in future.

File: 2nd-gen/packages/swc/components/close-button/test/close-button.test.ts (hunk 4 above)
Comment (TarunAdobe) [reply 2nd-gen/packages/swc/components/close-button/test/close-button.test.ts:426]: yessir

File: 2nd-gen/packages/swc/components/close-button/stories/close-button.stories.ts (hunk 3 above)
Comment (TarunAdobe) [reply 2nd-gen/packages/swc/components/close-button/stories/close-button.stories.ts]: fixed
